pkg/volume/host_path/host_path.go

```diff
@@ -211,7 +214,8 @@ func (b *hostPathMounter) SetUp(fsGroup *int64) error {
 	if err != nil {
 		return fmt.Errorf("invalid HostPath `%s`: %v", b.GetPath(), err)
 	}
-	return nil
+
+	return checkType(b.GetPath(), b.pathType, b.containerized)
```

Comment:
How this is setup, an extra `stat` call is made even for `v1.HostPathUnset`.

Having something like `if b.pathType == v1.HostPathUnset { return nil } else { return checkType... ` would avoid that.

`checkType` could also shortcut out sooner to avoid the stat.

I think it is worth avoiding that extra stat call in the default / backwards compatible case.